### PR TITLE
The URL for the fake server is now returned from `.spin`

### PR DESCRIPTION
**Pull request by jsom**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,10 +73,16 @@ some `MusicDB` library in the code that knows to hit the
 
 ```ruby
 FakeMusicDBRunner = Capybara::Discoball::Runner.new(FakeMusicDB) do |server|
-  MusicDB.endpoint_url = "http://#{server.host}:#{server.port}/"
+  MusicDB.endpoint_url = server.url
 end
 ```
 
+If no block is provided, the url is also returned by `#spin`:
```
Comment (mike-burns) [inline README.md]: All caps: "URL".

File: README.md [hunk 2]
```diff
@@ -73,10 +73,16 @@ some `MusicDB` library in the code that knows to hit the
...
 FakeMusicDBRunner = Capybara::Discoball::Runner.new(FakeMusicDB) do |server|
-  MusicDB.endpoint_url = "http://#{server.host}:#{server.port}/"
+  MusicDB.endpoint_url = server.url
 end
 ```
 
+If no block is provided, the URL is also returned by `#spin`:
+
+```ruby
+  MusicDB.endpoint_url = Capybara::Discoball.spin(FakeMusicDB)
```
Comment (gabebw) [inline README.md]: Why indent this line?

Comment (jsom) [inline README.md]: Fixed

File: lib/capybara_discoball/runner.rb [hunk 3]
```diff
@@ -11,11 +10,15 @@ def initialize(server_factory, &block)
 
       def boot
         with_webrick_runner do
-          @server = Capybara::Server.new(@server_factory.new)
+          @server = Server.new(@server_factory.new)
           @server.boot
         end
 
-        @after_server.call(@server)
+        if @after_server
```
Comment (gabebw) [inline lib/capybara_discoball/runner.rb]: Why is this `if` needed now?

File: lib/capybara_discoball/runner.rb [hunk 4]
```diff
@@ -11,11 +10,15 @@ def initialize(server_factory, &block)
...
+          @after_server.call(@server)
+        end
```
Comment (mike-burns) [inline lib/capybara_discoball/runner.rb]: If we have `#try` we could do:

```ruby
@after_server.try(:call, @server)
```

File: lib/capybara_discoball/runner.rb (hunk 3 above)
Comment (jsom) [inline lib/capybara_discoball/runner.rb]: You don't have to pass a block at all the Runner. Is there a nicer syntax to skip this call if there is no block?

File: lib/capybara_discoball/server.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+module Capybara
+  module Discoball
+    class Server < ::Capybara::Server
+      def url
+        "http://#{host}:#{port}"
+      end
+    end
+  end
+end
```
Comment (mike-burns) [inline lib/capybara_discoball/server.rb]: Thumbs up.

**Comment by mike-burns** [comment]
Yeah, LGTM.

File: lib/capybara_discoball/runner.rb [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
       end
 
       def boot
         with_webrick_runner do
-          @server = Capybara::Server.new(@server_factory.new)
+          @server = Server.new(@server_factory.new)
           @server.boot
         end
 
         @after_server.call(@server)
```
Comment (jsom) [inline lib/capybara_discoball/runner.rb]: @mike-burns @gabebw alternative approach?

Comment (mike-burns) [inline lib/capybara_discoball/runner.rb]: I love it.

**Comment by gabebw** [comment]
:+1:

**Pull request closed without merging**